**WACC (Weighted Average Cost of Capital)**

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
#WACC = (Weight of Equity × Cost of Equity) + (Weight of Debt × Cost of Debt × (1 − Tax Rate))

In [ ]:
rf = 4.5/100
market_premium = 5.5/100
beta = 0.92
tax_rate = 17.9/100
market_cap = 450
total_debt = 16.26
interest_expense = 0.575

In [ ]:
growth_rate = 0.1
terminal_rate = 2.5/100
fcf = 14.33
cash = 8.59

In [ ]:
cost_of_equity = rf + (beta*market_premium)
cost_of_debt = interest_expense/total_debt

In [ ]:
weight_of_equity = market_cap / (market_cap + total_debt)
weight_of_debt = total_debt / (market_cap + total_debt)

In [ ]:
wacc = (weight_of_equity*cost_of_equity) + (weight_of_debt*(cost_of_debt*(1-tax_rate)))
wacc

0.09327858919915925

In [ ]:
fcf_list = []
projected_fcf = fcf
for year in range(1,6):
  projected_fcf = fcf*(1 + growth_rate)**(year)
  fcf_list.append(projected_fcf)
  print(f"Projected fcf is {projected_fcf} billion dollars")

Projected fcf is 15.763000000000002 billion dollars
Projected fcf is 17.3393 billion dollars
Projected fcf is 19.073230000000006 billion dollars
Projected fcf is 20.980553000000008 billion dollars
Projected fcf is 23.07860830000001 billion dollars


In [ ]:
pv_list = []
pv = fcf

for year in range(1,6):
  pv = fcf_list[year-1]/(1 + wacc)**(year)
  pv_list.append(pv)
  print(f"Projected pv is {pv} billion dollars")

Projected pv is 14.418099975365477 billion dollars
Projected pv is 14.50674158406378 billion dollars
Projected pv is 14.595928156023962 billion dollars
Projected pv is 14.685663041647269 billion dollars
Projected pv is 14.775949611933019 billion dollars


In [ ]:
terminal_value = fcf_list[-1]*(1 + terminal_rate)/(wacc - terminal_rate)
terminal_value

346.4566826139298

In [ ]:
pv_of_tv = terminal_value/(1 + wacc)**5
pv_of_tv

221.8169492057671

In [ ]:
enterprise_value = sum(pv_list) + pv_of_tv
enterprise_value

294.7993315748006

In [ ]:
equity_value = enterprise_value - total_debt + cash
equity_value

287.1293315748006

DCF Valuation — Mastercard (2023)

FCF (2023): $14.33B
WACC: 9.33%
5-year FCF growth: 10%
Terminal growth rate: 2.5%
Enterprise Value: $294.8B
Equity Value: $287.1B
Market Cap (actual): ~$450B
Implied discount to market: ~36% — model is conservative; real valuation would incorporate higher growth assumptions and scenario analysis.

In [ ]:
wacc_range = np.arange(8,11.5,0.5)
terminal_rate = np.arange(1.5,4,0.5)

In [ ]:
sensitivity_analysis_matrix = pd.DataFrame(index=wacc_range, columns=terminal_rate)
sensitivity_analysis_matrix

,1.5,2.0,2.5,3.0,3.5
8.0,NaN,NaN,NaN,NaN,NaN
8.5,NaN,NaN,NaN,NaN,NaN
9.0,NaN,NaN,NaN,NaN,NaN
9.5,NaN,NaN,NaN,NaN,NaN
10.0,NaN,NaN,NaN,NaN,NaN
10.5,NaN,NaN,NaN,NaN,NaN
11.0,NaN,NaN,NaN,NaN,NaN


In [ ]:
for i in wacc_range:
    i_decimal = i / 100
    pv_list = [fcf / (1 + i_decimal)**t for t, fcf in enumerate(fcf_list, 1)]

    for j in terminal_rate:
        j_decimal = j / 100
        tv = fcf_list[-1] * (1 + j_decimal) / (i_decimal - j_decimal)
        pv_of_tv = tv / (1 + i_decimal)**5
        enterprise_value = sum(pv_list) + pv_of_tv
        equity_value = enterprise_value - total_debt
        sensitivity_analysis_matrix.loc[i, j] = equity_value

sensitivity_analysis_matrix

,1.5,2.0,2.5,3.0,3.5
8.0,304.739704,326.487737,352.189959,383.032624,420.729216
8.5,280.967688,299.267612,320.617522,345.849235,376.12729
9.0,260.379702,275.950624,293.917072,314.877928,339.649849
9.5,242.378416,255.755821,271.044285,288.68482,309.265444
10.0,226.507059,238.0975,251.233333,266.245714,283.567692
10.5,212.410372,222.527775,233.909854,246.809544,261.552046
11.0,199.807894,208.6983,218.634636,229.813014,242.481842
